update-price-table/update-price-table.py

```diff
@@ -32,30 +37,99 @@ def get_database_connection() -> connection:
         return error
 
 
-def get_product_data(rds_conn:connection):
-    """Query database for data on all products that price data is required for"""
+def create_ses_client():
+    """
+    Create and return a Boto3 client for AWS SES using AWS credentials.
+    """
+    ses_client = boto3.client(
+        'ses',
+        aws_access_key_id=environ["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=environ["AWS_SECRET_ACCESS_KEY"],
+        region_name="eu-west-2"
+    )
+    return ses_client
+
+
+def get_all_product_data(rds_conn:connection):
+    """
+    Query database for data on all products that price data is required for
+    """
 
     cur = rds_conn.cursor(cursor_factory=extras.RealDictCursor)
 
     # Get all unique products from product table
-    cur.execute("SELECT product_id, product_url FROM products;")
+    cur.execute("SELECT * FROM products;")
     rows = cur.fetchall()
     cur.close()
   
     return rows
 
 
-def scrape_asos_page(url: str, header: dict) -> dict:
+def get_user_emails(rds_conn:connection, product_id:int):
+    """
+    Query database for users which are subscribed to given product
+    """
+
+    cur = rds_conn.cursor(cursor_factory=extras.RealDictCursor)
+
+    cur.execute("""
+                SELECT users.email FROM users 
+                FULL OUTER JOIN subscriptions ON users.user_id = subscriptions.user_id 
+                WHERE subscriptions.product_id = %s
+                """,(product_id,))
+    rows = cur.fetchall()
+    cur.close()
+    
+    matching_emails = []
+    for entry in rows:
+        matching_emails.append(entry['email'])
+    
+    return matching_emails
+
+
+def send_unsubscription_emails(rds_conn:connection, ses_client:boto3.client, product_data):
+    """
+    Set sender, recipient and condition for email
+    """
+
+    sender = 'trainee.harvind.grewal@sigmalabs.co.uk'
+    recipients = get_user_emails(rds_conn, product_data['product_id'])
+
+    ## Include email verification upon email entry on webpage ##
+    for recipient in recipients:
+        response = ses_client.verify_email_identity(
+            EmailAddress=recipient
+        )
+    ## Include email verification upon email entry on webpage ##
+        
+    for recipient in recipients:
+        response = ses_client.send_email(
+            Source=sender,
+            Destination={'ToAddresses': [recipient]},
+            Message={
+                'Subject': {'Data': "Notification of product removed from tracking"},
+                'Body': {'Text': {'Data': f"""
+                                The product ({product_data['product_name']}) you have been tracking is no longer accessible by our system via the link provided 
+                                ({product_data['product_url']}). Please visit our website to re-subscribe a product
+                                """}}
+            }
+        )
+
```

Comment:
Pretty email formatting could be cool here? Not essential but may make the product more impressive/professional to clients.